packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/tags/GetTagsTests.java

```diff
@@ -226,4 +228,91 @@ public void checkThatDeletedTagIsNotRetrievedAnymore() throws Exception
+            .entriesListContains("tag", documentTagValue.toLowerCase());
+    }
+
+    /**
+     * Get tags with names filter using IN and expect two items in result.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_withTwoNameFilters()
+    {
+        STEP("Get tags with names filter using IN and expect two items in result");
+        returnedCollection = restClient.authenticateUser(adminUserModel)
+            .withParams("where=(name IN ('" + documentTag.getTag() + "', '" + folderTag.getTag() + "'))")
+            .withCoreAPI()
+            .getTags();
+
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+        returnedCollection.assertThat()
+            .entriesListCountIs(2).and()
+            .entriesListContains("tag", documentTagValue.toLowerCase()).and()
+            .entriesListContains("tag", folderTagValue.toLowerCase());
+    }
+
+    /**
+     * Get tags with names filter using EQUALS (in fact contains) and expect two item in result.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_whichNamesStartsWithOrphan()
+    {
+        STEP("Get tags with names filter using EQUALS (in fact contains) and expect two item in result");
+        returnedCollection = restClient.authenticateUser(adminUserModel)
+            .withParams("where=(name=orphan)")
+            .withCoreAPI()
+            .getTags();
+
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+        returnedCollection.assertThat()
+            .entriesListIsNotEmpty().and()
+            .entriesListContains("tag", orphanTag.getTag().toLowerCase());
+    }
+
+    /**
+     * Try to get tags with names filter using EQUALS and wrong property name and expect 400 (Bad Request).
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_withWrongWherePropertyNameAndExpect400()
+    {
+        STEP("Try to get tags with names filter using EQUALS and wrong property name and expect 400");
+        returnedCollection = restClient.authenticateUser(adminUserModel)
+            .withParams("where=(tag=gat)")
+            .withCoreAPI()
+            .getTags();
+
+        restClient.assertStatusCodeIs(HttpStatus.BAD_REQUEST)
+            .assertLastError().containsSummary("Where query error: property with name: tag is not expected");
+    }
+
+    /**
+     * Try to get tags with names filter using EQUALS and IN at the same time and expect 400 (Bad Request).
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_bothEqualsAndInNotAllowed()
+    {
+        STEP("Try to get tags with names filter using EQUALS and IN at the same time and expect 400");
+        returnedCollection = restClient.authenticateUser(adminUserModel)
+            .withParams("where=(name=tag AND name IN ('tag-', 'gat'))")
```

Comment:
Is there a technical reason why we can't use these together? Can we use `where=(name=tag AND name=tag)`?  I would have expected it to work, but I don't think it's very useful functionality to double filter the field like this.

Reply:
The issue is that I don't know which filter should be picked up for e.g. `where=(tag=hr AND tag=is)`. We cannot determine if tag should be "hr" or "is". This is the reason `AND` is not supported for tags filtering. I think that we can support `OR` operator instead.

Reply:
I think that `where=(tag=hr AND tag=is)` should match no tags. Anyway - I think we discussed already that it's fine not to support boolean operators at all if they're not trivial. The main feature we want is for the FE to be able to filter by exact match (`where=(tag in ('hr', 'is'))`).

Reply:
Hmm... 🤔 good point with no tags! Anyway `AND` is not supported. `OR` wasn't too complex, so it's supported.